packages/asset-kit-block/src/AssetKitBlock.tsx

```diff
@@ -39,24 +79,35 @@ export const AssetKitBlock = ({ appBridge }: BlockProps): ReactElement => {
                 <div className="tw-flex-none">
                     <button
                         data-test-id="asset-kit-block-download-button"
-                        disabled={isDownloadingAssets || isUploadingAssets || currentAssets.length <= 0}
-                        onClick={() =>
-                            generateBulkDownload(appBridge.getProjectId(), currentAssets, setIsDownloadingAssets)
+                        disabled={
+                            [BulkDownloadState.Error, BulkDownloadState.Pending, BulkDownloadState.Started].includes(
+                                status
+                            ) || currentAssets.length <= 0
```

Comment:
Minor, but I think this makes it a tiny bit easier to understand

```suggestion
                            ) || currentAssets.length === 0
```